packages/calling/src/Voicemail/BroadworksBackendConnector.test.ts

```diff
@@ -306,8 +413,69 @@ describe('Voicemail Broadworks Backend Connector Test case', () => {
       sessionStorage.removeItem(CONTEXT);
     });
 
+    it('verify empty voicemail listing data', async () => {
+      global.fetch.mockReturnValueOnce({
+        status: 200,
+        ok: true,
+        json: () => Promise.resolve(getEmptyVoicemailListJsonBWRKS),
+      });
+
+      const response = await broadworksBackendConnector.getVoicemailList(0, 20, SORT.DESC, true);
+
+      const voicemailResponseInfo = {
+        voicemailList: [],
+      };
+
+      const responseDetails = {
+        statusCode: NO_VOICEMAIL_STATUS_CODE,
+        data: voicemailResponseInfo,
+        message: NO_VOICEMAIL_MSG,
+      };
+
+      expect(response).toStrictEqual(responseDetails);
+      expect(response.message).toBe(NO_VOICEMAIL_MSG);
+      expect(global.fetch).toBeCalledOnceWith(`${broadworksUserInfoUrl}${JSON_FORMAT}`, {
+        headers: {Authorization: `bearer ${bwToken}`},
+        method: 'GET',
+      });
+      expect(fetchVoicemailListSpy).toBeCalledOnceWith(CONTEXT, 0, 20, {
+        file: 'BroadworksBackendConnector',
+        method: 'getVoicemailList',
+      });
+    });
+
+    it('verify empty voicemail listing data when response data is in invalid format', async () => {
+      global.fetch.mockReturnValueOnce({
+        status: 200,
+        ok: true,
+        json: () => Promise.resolve(getInvalidVoicemailListJsonBWRKS),
+      });
+
+      const response = await broadworksBackendConnector.getVoicemailList(0, 20, SORT.DESC, true);
+
+      const voicemailResponseInfo = {
+        voicemailList: [{}],
+      };
+
+      const responseDetails = {
+        statusCode: NO_VOICEMAIL_STATUS_CODE,
+        data: voicemailResponseInfo,
```

Comment:
In the above test an empty voicemailResponseInfo is initialised to [] and here we have [{}]. Arent they the same array?

Reply:
No, they are testing different conditions:
1. Empty array is testing this condition: https://github.com/webex/webex-js-sdk/blob/next/packages/calling/src/Voicemail/BroadworksBackendConnector.ts#L235

2. [{}] is testing this: https://github.com/webex/webex-js-sdk/blob/next/packages/calling/src/Voicemail/BroadworksBackendConnector.ts#L240